src/gui/widget_click.c

```diff
@@ -42,8 +42,8 @@ static uint16 s_savegameCountOnDisk = 0;                    /*!< Amount of saveg
 
 static char *GenerateSavegameFilename(uint16 number)
 {
-	static char filename[13];
-	sprintf(filename, "_save%03d.dat", number);
+	static char filename[15];
+	sprintf(filename, "_save%03hx.dat", number);
```

Comment:
it should be %03hu !!!

Reply:
and filename is 8.3 for sure ! use snprintf() instead of increasing filename buffer

Reply:
Correct, I messed this one up, it should be "%03hu". Note that since even if we specify that the number should be prefixed by 3 zeroes, if you attempt to print 65535, it will output that, not 535 so that's why I had to increase the output buffer again.

Reply:
but number should never be > 999

Reply:
either use snprint and/or clamp number to not be bigger than 999 ?